src/when.js

```diff
@@ -33,7 +33,7 @@ const checkArgumentMatchers = (expectCall, args) => (match, matcher, i) => {
   }
 
   if (isFunctionMatcher) {
-    return matcher(arg)
+    return matcher(arg, utils.equals)
```

Comment:
Is this new functionality for the regular function matchers, too? Seems nice to have, and (if I'm not reading this incorrectly) might be worth covering with a test-case in the regular `when(matcher)` context

Reply:
good point, yeah I just realized folks might like the power of that equals util in their own custom function matchers. I'll add a test.